apps/prairielearn/src/lib/groups.js

```diff
@@ -458,6 +458,14 @@ module.exports.updateGroupRoles = async function (requestBody, assessmentId, use
         "Couldn't access the user's group ID with the provided assessment and user IDs"
       );
     }
+
+    const permissions = await module.exports.getAssessmentPermissions(assessmentId, userId);
+    if (!permissions.can_assign_roles_at_start) {
+      throw new Error(
+        'User does not have permission to assign roles at the start of this assessment'
+      );
```

Comment:
This should throw an error with a 403 status code. This is both more accurate *and* keeps this kind of error from showing up in our Sentry reporting of unexpected errors. Search for `error.make(403` in the codebase to see other examples.